src/Symfony/Component/Messenger/Exception/MessageRecorderHandlerException.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Exception;
+
+/**
+ * When handling recorded messaged one or more handlers caused an exception.
+ * This exception contains all those handler exceptions.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class MessageRecorderHandlerException extends \RuntimeException implements ExceptionInterface
+{
+    private $exceptions = array();
+
+    public function __construct(array $exceptions)
+    {
+        $message = sprintf(
+            "One or more handlers for recorded messages threw an exception. Their messages were: \n\n%s",
```

Comment:
If there is only one exception, let's not confuse the user and display it without wrapping it to "one or more".